### PR TITLE
Fix github API errors in check_style CI jobs

### LINKED ISSUE #10190: Fix scripts/check_style.sh to work with latest Github API
From https://app.circleci.com/pipelines/github/ethereum/solidity/9790/workflows/c58049a5-b961-4e6b-8ceb-4d47fa3a7168/jobs/487386:
```
#!/bin/bash -eo pipefail
./scripts/check_style.sh
Error: Trailing whitespace found:
test/libsolidity/semanticTests/underscore/as_function.sol:5:    
test/libsolidity/semanticTests/underscore/in_modifier.sol:11:    
test/libsolidity/syntaxTests/underscore/as_function.sol:4:    
test/libsolidity/syntaxTests/underscore/in_modifier.sol:11:    
Error: Trailing whitespace found: test/libsolidity/semanticTests/underscore/as_function.sol:5: test/libsolidity/semanticTests/underscore/in_modifier.sol:11: test/libsolidity/syntaxTests/underscore/as_function.sol:4: test/libsolidity/syntaxTests/underscore/in_modifier.sol:11:
posting error message to github
{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest"
}
{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest"
}
{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest"
}
{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest"
}
{
  "message": "Not Found",
  "documentation_url": "https://docs.github.com/rest"
}

Exited with code exit status 1
CircleCI received exit code 1
```

### PR DESCRIPTION
**Pull request by cameel**
Fixes #10190.

### CODE REVIEW

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `f256cf3f41aab3686163da40f8bcbb8104c45225`:
```
Coding style error:
 libsolidity/codegen/ABIFunctions.cpp:66: if (_reversed) {

```
Please check that your changes are working as intended.

File: libsolidity/codegen/ABIFunctions.cpp [hunk 1]
```diff
@@ -63,8 +63,9 @@ string ABIFunctions::tupleEncoder(
 	for (auto const& t: _targetTypes)
 		functionName += t->identifier() + "_";
 	functionName += options.toFunctionNameSuffix();
-	if (_reversed)
+	if (_reversed) {
```
Comment (stackenbotten) [inline libsolidity/codegen/ABIFunctions.cpp]: Coding style error

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `f256cf3f41aab3686163da40f8bcbb8104c45225`:
```
libsolidity/codegen/ABIFunctions.cpp:66: if (_reversed) {
 Coding style error:
 libsolidity/codegen/ABIFunctions.cpp:66: if (_reversed) {

```
Please check that your changes are working as intended.

File: libsolidity/codegen/ABIFunctions.cpp (hunk 1 above)
Comment (stackenbotten) [inline libsolidity/codegen/ABIFunctions.cpp]: Coding style error

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `f256cf3f41aab3686163da40f8bcbb8104c45225`:
```
Coding style error:
 libsolidity/codegen/ABIFunctions.cpp:66: if (_reversed) {
 Coding style error:
 libsolidity/codegen/ABIFunctions.cpp:66: if (_reversed) {

```
Please check that your changes are working as intended.

File: libsolidity/codegen/ABIFunctions.cpp (hunk 1 above)
Comment (stackenbotten) [inline libsolidity/codegen/ABIFunctions.cpp]: Coding style error

Comment (stackenbotten) [inline libsolidity/codegen/ABIFunctions.cpp]: Coding style error

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `f256cf3f41aab3686163da40f8bcbb8104c45225`:
```
libsolidity/codegen/ABIFunctions.cpp:66: if (_reversed) {
 Coding style error:
 libsolidity/codegen/ABIFunctions.cpp:66: if (_reversed) {

```
Please check that your changes are working as intended.

File: libsolidity/codegen/ABIFunctions.cpp (hunk 1 above)
Comment (stackenbotten) [inline libsolidity/codegen/ABIFunctions.cpp]: Coding style error

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `f256cf3f41aab3686163da40f8bcbb8104c45225`:
```
libsolidity/codegen/ABIFunctions.cpp:66: if (_reversed) {

```
Please check that your changes are working as intended.

File: libsolidity/codegen/ABIFunctions.cpp (hunk 1 above)
Comment (stackenbotten) [inline libsolidity/codegen/ABIFunctions.cpp]: Coding style error

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `f256cf3f41aab3686163da40f8bcbb8104c45225`:
```
Coding style error:
 libsolidity/codegen/ABIFunctions.cpp:66: if (_reversed) {

```
Please check that your changes are working as intended.

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `f256cf3f41aab3686163da40f8bcbb8104c45225`:
```
libsolidity/codegen/ABIFunctions.cpp:66: if (_reversed) {

```
Please check that your changes are working as intended.

File: libsolidity/codegen/ABIFunctions.cpp (hunk 1 above)
Comment (stackenbotten) [inline libsolidity/codegen/ABIFunctions.cpp]: Coding style error

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `f256cf3f41aab3686163da40f8bcbb8104c45225`:
```
Coding style error:
 libsolidity/codegen/ABIFunctions.cpp:66: if (_reversed) {

```
Please check that your changes are working as intended.

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `f256cf3f41aab3686163da40f8bcbb8104c45225`:
```
libsolidity/codegen/ABIFunctions.cpp:66: if (_reversed) {
 Coding style error:
 libsolidity/codegen/ABIFunctions.cpp:66: if (_reversed) {

```
Please check that your changes are working as intended.

File: libsolidity/codegen/ABIFunctions.cpp (hunk 1 above)
Comment (stackenbotten) [inline libsolidity/codegen/ABIFunctions.cpp]: Coding style error

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `f256cf3f41aab3686163da40f8bcbb8104c45225`:
```
Coding style error:
 libsolidity/codegen/ABIFunctions.cpp:66: if (_reversed) {
 Coding style error:
 libsolidity/codegen/ABIFunctions.cpp:66: if (_reversed) {

```
Please check that your changes are working as intended.

File: libsolidity/codegen/ABIFunctions.cpp (hunk 1 above)
Comment (stackenbotten) [inline libsolidity/codegen/ABIFunctions.cpp]: Coding style error

Comment (stackenbotten) [inline libsolidity/codegen/ABIFunctions.cpp]: Coding style error

Comment (stackenbotten) [inline libsolidity/codegen/ABIFunctions.cpp]: Coding style error

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `f256cf3f41aab3686163da40f8bcbb8104c45225`:
```
Coding style error:
 libsolidity/codegen/ABIFunctions.cpp:66: if (_reversed) {

```
Please check that your changes are working as intended.

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `d2de9d90e986fbf5041b7d83d765a802a21fabb8`:
```
Coding style error:
 libsolidity/codegen/ABIFunctions.cpp:66: if (_reversed) {

```
Please check that your changes are working as intended.

File: libsolidity/codegen/ABIFunctions.cpp (hunk 1 above)
Comment (stackenbotten) [inline libsolidity/codegen/ABIFunctions.cpp]: Coding style error

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `f256cf3f41aab3686163da40f8bcbb8104c45225`:
```
Coding style error:
 libsolidity/codegen/ABIFunctions.cpp:66: if (_reversed) {

```
Please check that your changes are working as intended.

File: libsolidity/codegen/ABIFunctions.cpp (hunk 1 above)
Comment (stackenbotten) [inline libsolidity/codegen/ABIFunctions.cpp]: Coding style error

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `98462dbc5683dbb60a8797a80443e6fd3573a94e`:
```
Coding style error:
 libsolidity/codegen/ABIFunctions.cpp:66: if (_reversed) {

```
Please check that your changes are working as intended.

File: libsolidity/codegen/ABIFunctions.cpp (hunk 1 above)
Comment (stackenbotten) [inline libsolidity/codegen/ABIFunctions.cpp]: Coding style error

**Comment by stackenbotten** [comment]
There was an error when running `chk_coding_style` for commit `72e9ea77cd66f326c6625f9281c5c4b9e1af9fbe`:
```
Coding style error:
 libsolidity/codegen/ABIFunctions.cpp:66: if (_reversed) {

```
Please check that your changes are working as intended.

File: libsolidity/codegen/ABIFunctions.cpp (hunk 1 above)
Comment (stackenbotten) [inline libsolidity/codegen/ABIFunctions.cpp]: Coding style error

File: scripts/report_errors.sh [hunk 2]
```diff
@@ -1,14 +1,9 @@
 #!/usr/bin/env bash
 
-export ERROR_LOG="/tmp/error.log"
+ERROR_LOG="$1"
 
 function report_error_to_github
 {
-    if [ $? -eq 0 ]
```
Comment (axic) [inline scripts/report_errors.sh:7]: Why not just `set -e` at the top?

Comment (cameel) [reply scripts/report_errors.sh:7]: I think that the original reason might have been that this script used to be sourced by `check_style.sh` which uses grep a lot. grep returns 1 if there's no match.

But in this PR I wanted to include only the changes actually needed to fix the errors, without any extra refactors, to get it out today and minimize the the chance of introducing new problems. It's basically ready, I just need to remove the dummy commit with a style error.

I'm also preparing another PR with a more complete refactor of these two scripts (including fixes for shellcheck warnings from my pending draft PRs) and I'll try to use `set -e` there. But it's getting late so I think I'll finish that on Monday.

Comment (cameel) [reply scripts/report_errors.sh:7]: Actually I decided to just post what I have in a separate PR: #10797. No `set -e` yet though :)

I'll go back to this later, when I'll be finalizing drafts of my other shellcheck-releated PRs (that are basically ready and just need some extra testing tomake sure they don't break stuff).

Comment (leonardoalt) [reply scripts/report_errors.sh:7]: @cameel there's a cyclic PR dependency here: you said above you want to do #10797  before this here. But #10797 says "NOTE: This PR is based on #10792 ", which in this one. So what's the order?

Comment (cameel) [reply scripts/report_errors.sh:7]: This one (#10792) is first - it contains only the minimum necessary to fix the errors in CI.

#10797 is on top of it and contains only refactors that I managed to do on Friday. Unfortunately that does not include `set -e`.

**Review by leonardoalt — APPROVED** [review]

**Pull request merged**